compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1933,28 +1933,90 @@ TR::Register *OMR::X86::TreeEvaluator::integerDualMulEvaluator(TR::Node *node, T
 
 TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR::Register *targetRegister = 0;
-   TR::Node     *firstChild     = node->getFirstChild();
-   TR::Node     *secondChild    = node->getSecondChild();
-   bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
-   intptrj_t    value;
+   TR::Register *targetRegister  = 0;
+   TR::Node     *firstChild      = node->getFirstChild();
+   TR::Node     *secondChild     = node->getSecondChild();
+   bool          nodeIs64Bit     = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool          isLoadConst     = secondChild->getOpCode().isLoadConst();
+   bool          isByteOperation = false;
+   bool          isMemRef        = false;
+   TR_X86OpCodes opCode          = BADIA32Op; 
+   TR_X86OpCodes regOpCode       = BADIA32Op; 
+   TR_X86OpCodes memOpCode       = BADIA32Op; 
+   TR_X86OpCodes movOpCode       = BADIA32Op;
+   int64_t       value;
 
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
 
-   if (secondChild->getOpCode().isLoadConst() &&
-      (value = TR::TreeEvaluator::integerConstNodeValue(secondChild, cg)))
+   // Opcode evaluation for bmul, smul imul and lmul
+   switch (secondChild->getDataType())
+      {
+      case TR::Int8:
+         isMemRef = secondChild->getOpCode().isMemoryReference(); 
+         opCode = isMemRef ? IMUL1AccMem : IMUL1AccReg;
+     
+         regOpCode = IMUL1AccReg;
```

Comment:
I personally prefer to move the declaration and assignment of regOpCode, memOpCode, movOpCode closer to the place using them inside  if (targetRegister == 0). These are only used in the analyser when all the optimizations failed.